src/pages/index.js

```diff
@@ -63,120 +35,140 @@ const editModalDescriptionInput = editModal.querySelector(
 avatarModalBtn.addEventListener("click", () => {
   resetValidation(avatarForm, settings);
   const submitButton = avatarForm.querySelector(".modal__submit-btn");
   disabledButton(submitButton, settings);
   openModal(avatarModal);
 });
 
-// Close the avatar modal when the close button is clicked
-avatarModalCloseBtn.addEventListener("click", () => {
-  closeModal(avatarModal);
-});
-
-// Add submit event listener
-avatarForm.addEventListener("submit", (evt) => {
-  console.log("Form submit triggered"); // Debugging log
-  handleAvatarSubmit(evt);
-});
+avatarModalCloseBtn.addEventListener("click", () => closeModal(avatarModal));
+previewModalCloseBtn.addEventListener("click", () => closeModal(previewModal));
+cardModalCloseBtn.addEventListener("click", () => closeModal(cardModal));
+editModalCloseBtn.addEventListener("click", () => closeModal(editModal));
+deleteModalCloseBtn.addEventListener("click", () => closeModal(deleteModal));
+deleteModalCancelBtn.addEventListener("click", () => closeModal(deleteModal));
 
-function handleAvatarSubmit(evt) {
-  console.log("Inside handleAvatarSubmit"); // Debugging log
-  evt.preventDefault();
+// ============ Form Submissions ============
 
+avatarForm.addEventListener("submit", (e) => {
+  e.preventDefault();
   const avatarUrl = avatarInput.value;
-  console.log("Avatar URL:", avatarUrl); // Debugging log to check input value
 
   api
     .editAvatarInfo({ avatar: avatarUrl })
     .then((data) => {
-      if (data && data.avatar) {
-        profileAvatar.src = data.avatar; // Ensure you're updating the `src` attribute
-      } else {
-        console.error("API response does not contain avatar URL:", data);
-      }
+      profileAvatar.src = data.avatar;
       closeModal(avatarModal);
     })
-    .catch((err) => {
-      console.error("Error saving avatar:", err);
-    });
-}
+    .catch((err) => console.error("Error saving avatar:", err));
+});
 
-function handleDeleteCard(evt) {
-  evt.target.closest(".card").remove();
-  openModal(deleteModal);
-}
+editFormElement.addEventListener("submit", (e) => {
+  e.preventDefault();
+  const submitButton = editFormElement.querySelector(".modal__submit-btn");
+  setLoadingState(submitButton, true, "Saving...");
 
-function getCardElement(data) {
-  const cardElement = cardTemplate.content
-    .cloneNode(true)
-    .querySelector(".card");
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((userData) => {
+      profileName.textContent = userData.name;
+      profileDescription.textContent = userData.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => setLoadingState(submitButton, false, "Save"));
+});
 
-  const cardNameEl = cardElement.querySelector(".card__title");
-  const cardImageEl = cardElement.querySelector(".card__image");
+cardForm.addEventListener("submit", (e) => {
+  e.preventDefault();
 
```

Comment:
* When I add a new card or change the `avatar`, the button text doesn't change to `Saving...`